v03_pipeline/lib/reference_datasets/hgmd.py

```diff
@@ -0,0 +1,30 @@
+import hail as hl
+
+from v03_pipeline.lib.model import ReferenceGenome
+
+HGMD_CLASSES = [
+    'DM',
+    'DM?',
+    'DP',
+    'DFP',
+    'FP',
+    'R',
+]
+
+
+def get_ht(raw_dataset_path: str, reference_genome: ReferenceGenome) -> hl.Table:
+    mt = hl.import_vcf(
+        raw_dataset_path,
+        reference_genome=reference_genome.value,
+        force=True,
+        min_partitions=100,
```

Comment:
I haven't been setting an explicit min_partitions and have just need relying on hail to pick a reasonable number. Is there a way we should go about picking a partition number?

Reply:
Good question, optimal partitioning remains somewhat of a mystery to me, and in this case since we're using `force=True` everything will end up on one partition (I could be wrong) so maybe the move here is to repartition after import.

It looks like there is an optimal partition size of 128 mb, maybe we should start using this more: https://github.com/broadinstitute/seqr-loading-pipelines/blob/main/v03_pipeline/lib/misc/io.py#L72

Reply:
I think hail will default to the number of partitions in the sharded vcf, which is 1 here (because it's not sharded).  The partitioning will only matter through the import step, as we'll repartition when we call `write` to [write the file](https://github.com/broadinstitute/seqr-loading-pipelines/blob/main/v03_pipeline/lib/misc/io.py#L293).  

But yes, `force=True` forces serial execution, so the `min_partitions` is useless:

```
In [7]: mt = hl.import_vcf('gs://seqr-reference-data-private/GRCh37/HGMD/HGMD_Pro_2023.1_hg19.vcf.gz', force=True, min_partitions=100)
2024-11-14 17:47:43.245 Hail: WARN: You are trying to read gs://seqr-reference-data-private/GRCh37/HGMD/HGMD_Pro_2023.1_hg19.vcf.gz with *ONE* core of parallelism. This will be very slow. If this file is block-gzipped (bgzip-ed), use force_bgz=True instead.

In [8]: mt.n_partitions()
Out[8]: 1
```

Reply:
From the hail docs:
```
force ([bool](https://docs.python.org/3.9/library/functions.html#bool)) – If True, load gzipped files serially on one core. This should be used only when absolutely necessary, as processing time will be increased due to lack of parallelism.
```
Is it absolutely necessary here?

Reply:
> It looks like there is an optimal partition size of 128 mb, maybe we should start using this more: https://github.com/broadinstitute/seqr-loading-pipelines/blob/main/v03_pipeline/lib/misc/io.py#L72

Is there any reason not to run this on all the tables we generate before we write out the final file?

Reply:
^ it's running already on every call to `create_table`.

Reply:
I removed `min_partitions`